src/sigs.ml

```diff
@@ -8,11 +8,12 @@ module type IO = sig
   val ( let* ) : 'a t -> ('a -> 'b t) -> 'b t
   val ( and+ ) : 'a t -> 'b t -> ('a * 'b) t
 
+  type env
   type in_channel
   type out_channel
 
-  val stdin : in_channel
-  val stdout : out_channel
+  val stdin : env -> in_channel
```

Comment:
it's not required for eio right now, but maybe env should be passed to read/read_line/write? is there a chance we'll need it with a different eio API ?

Reply:
I think the main thing is that the type of the writing functions requires changing if we want to support buffered writes (as in https://github.com/ocaml-multicore/eio?tab=readme-ov-file#buffered-writing).

I'm not sure if this is worth doing, but otherwise I believe the API as used here is quite stable.